meta-lmp-base/recipes-support/compose-apps-early-start/compose-apps-early-start/compose-apps-early-start-recovery

```diff
@@ -21,6 +22,7 @@ function recover_restorable_apps() {
 	systemctl restart compose-apps-register-images.service
 	systemctl start docker
 	systemctl reset-failed compose-apps-early-start.service
+	sleep 60
```

Comment:
What is the consequence if we get a broken app which can cause this job to be executed in loop? I guess lmp will be running but docker will be constantly restarting on an infinite loop.

Reply:
That's correct. At least a device won't be constantly restarting.
Unfortunately, I don't see a way to limit a number for attempts for `early-start` --> `early-start-recovery` --> `early start` cycle. We could limit it if we use `ExecStopPost` to invoke the recovery script.

Reply:
Right, the problem is that at this stage the device is not even registered, so there won't be much that the user can do with it (assuming login is disabled, for example).

Having it rebooting could end up causing the recovery process to be initialized (once we enable it), which might be better. I understand that rebooting is not really helpful, but not rebooting is not a lot better.

I'm ok to keep the sleep for now, until we enable the recovery logic, we just need to make sure that this limitation gets documented somehow (@kprosise). Basically we need to explicitly say that the apps are required to be known to work and not conflict with each other, otherwise the device will be in a bad state.

Reply:
> that at this stage the device is not even registered

Well, it depends, I can imagine a scenario when a device is preregistered before its first boot.

> so there won't be much that the user can do with it 

Yes, as I stated earlier, this workaround does not help end users much because their device does not function properly anyway.
However, I find it is useful for customers' engineers during dev and test phases of Apps development. It would be hard/inconvenient for them to figure out App failure reason if a device keeps rebooting.

It's just a workaround and a proper solution I think is out of the scope of the given PR/effort. IMO, a proper solution should limit a number of attempts to two, if it doesn't work after starting from scratch then it doesn't make any sense to keep trying.